torchvision/io/video_reader.py

```diff
@@ -173,6 +237,6 @@ def set_current_stream(self, stream: str) -> bool:
         Returns:
             (bool): True on succes, False otherwise
         """
-        if self.is_cuda:
+        if self.backend == "cuda":
             print("GPU decoding only works with video stream.")
```

Comment:
I think this should be changed to a warning. But can be done on a different PR as it is out of scope

Reply:
Done